src/components/BottomBar/GeoPositionPanel.jsx

```diff
@@ -0,0 +1,310 @@
+  const [altitude, setAltitude] = useLocalStorageState('altitude', 300000);
+  const [interaction, setInteraction] = useLocalStorageState('interaction', Interaction.flyTo);
+  const [currentAnchor, setCurrentAnchor] = useLocalStorageState('anchor', 'Earth');
+  const options = ['Earth', "Mars", "Test"];
+
+  function getPlaces()  {
+    const searchString = inputValue.replaceAll(" ", "%");
+    fetch(`https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/findAddressCandidates?SingleLine=${searchString}&category=&outFields=*&forStorage=false&f=json`)
+        .then(response => response.json())
+        .then(json => {
+          // Remove duplicates
+          const uniqueLabels = [];
+          const unique = json?.candidates.filter(place => {
+            const isDuplicate = uniqueLabels.includes(place.attributes.LongLabel);
+
+            if (!isDuplicate) {
+              uniqueLabels.push(place.attributes.LongLabel);
+              return true;
+            }
+            return false;
+          });
+          setPlaces(unique)
+        })
+        .catch(error => console.log("Error fetching: ", error));
+  };
+
+  function pushSceneGraphNode(nodeId) {
+    // Deep copy
+    const nodes = [...addedSceneGraphNodes];
+    nodes.push(nodeId);
+    setAddedSceneGraphNodes(nodes);
+  }
+
+  function togglePopover() {
+    setPopoverVisibilityProp(!popoverVisible);
+  }
+
+  function onClick(location, address, pushSceneGraphNode) {
+    const lat = location.y;
+    const long = location.x;
+    switch(interaction) {
+      case Interaction.flyTo: {
+        luaApi?.globebrowsing?.flyToGeo(currentAnchor, lat, long, altitude);
+        break;
+      }
+      case Interaction.jumpTo: {
+        luaApi?.globebrowsing?.goToGeo(currentAnchor, lat, long, altitude);
+        break;
+      }
+      case Interaction.addFocus: {
+        // Don't add if it is already added
+        if (addedSceneGraphNodes.indexOf(address) > -1) {
+          break;
+        }
+        pushSceneGraphNode(address);
+        let addressString = address;
+        addressString.replace(/[^\x00-\x7F]/g, "");
+        luaApi?.addSceneGraphNode(createSceneGraphNodeTable(currentAnchor, addressString, lat, long, altitude));
+        // TODO: Once we have a proper way to subscribe to additions and removals
+        // of property owners, this 'hard' refresh should be removed.
+        setTimeout(() => {
+          refresh();
+        });
+        break;
+      }
+      default: {
+        luaApi?.globebrowsing?.flyToGeo(currentAnchor, lat, long, altitude);
+        break;
+      }
+    }
+  }
+
+  function anchorPanel(anchor) {
+    switch (anchor) {
+      case 'Earth':
+        return <>
+            <div className={styles.searchField}>
+              <Input
+                placeholder={"Search places..."}
+                onKeyPress={(e) => {
+                  if (e.key === 'Enter') {
+                    getPlaces();
```

Comment:
This should be removed from here and done in the `onEnter` property of the `Input` component